sympy/matrices/tests/test_matrixbase.py

```diff
@@ -351,6 +351,16 @@ def test_replace_map():
     assert N == Matrix(2, 2, lambda i, j: G(i+j))
     assert d == {F(0): G(0), F(1): G(1), F(2): G(2)}
 
+def test_numpy_conversion():
+    try:
+        from numpy import array, array_equal
+    except ImportError:
+        skip('NumPy must be available to test creating matrices from ndarrays')
+    A = Matrix([[1,2], [3,4]])
+    np_array = array([[1,2], [3,4]])
+    assert array_equal(array(A), np_array)
+    assert array_equal(array(A, copy=True), np_array)
+    #raises(TypeError, lambda: array(A, copy=False)) TODO: Uncomment this whenever copy variable properly passes to __array__
```

Comment:
This works as intended if numpy 2.0.0rc1 is installed.

We should add a version check
```
int(importlib.metadata.version('numpy').split('.')[0]) >= 2
```